main/test/flix/experimental/Test.Dec.AssocType.flix

```diff
@@ -0,0 +1,8 @@
+mod Test/Dec/AssocType {
+
+    class Coll[a] {
+        type Elem[a]: Type
```

Comment:
What's the reasoning for this specific form? I would have thought that it would just be `type Elem` or maybe `type Elem: Type`.

Reply:
This syntax is more precise IMO. Note that `Elem` is not a `Type`, but rather a type-level function `Type -> Type`, which will be used like `Elem[String]`. It also ensures (in case we expand to multi-param associated types) that we know which parameter to use for lookup.